client/interfaces.go

```diff
@@ -27,7 +29,7 @@ type Client interface {
 
 // Writer is a client taht sends a batch of samples to remote.
 type Writer interface {
-	Write(samples model.Samples) error
+	Write(samples model.Samples, urlValues url.Values) error
```

Comment:
What about passing directly the HttpRequest ? This way the writer would have access to headers too.

Reply:
Because we just need the url.Values. Most of the time we are using r *http.Request.
Using r *http.Request here will require that we add the following lines which are already in the main.go:
```
urlValues, err := url.ParseQuery(r.URL.RawQuery)
	reqBuf, err := snappy.Decode(nil, compressed)
	if err != nil {
		level.Warn(logger).Log("err", err, "msg", "Error decoding request body")
		http.Error(w, err.Error(), http.StatusBadRequest)
		return
	}
```

Reply:
My point is that other clients my use other parts of the http request to do other things.
I'd prefer to pass the httprequest, and to add a helper method in client/graphite/ to help find the correct prefix.

If you do that, you can remove this code from main.go

Reply:
something like that is ok for you: 
```
if graphitePrefix, ok := urlValues["graphite.default-prefix"]; ok {
		graphitePrefix = urlValues["graphite.default-prefix"]
		return graphitePrefix[0], nil
	} else {
		graphitePrefix := c.cfg.DefaultPrefix
		return graphitePrefix, nil
	}
```

Reply:
you don't need to assign it to graphitePrefix in this example, but yes